code/modules/reagents/newchem/food.dm

```diff
@@ -133,17 +133,19 @@ datum/reagent/honey/reaction_turf(var/turf/T, var/volume)
 	reagent_state = LIQUID
 	color = "#2E2418"
 
-datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
+/datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
 	if(!M) M = holder.my_atom
 	M.reagents.add_reagent("sugar", 0.8)
 	..()
 	return
 
-datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
+/datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
 	src = null
-	if(volume >= 5)
-		new /obj/item/weapon/reagent_containers/food/snacks/reagentchocolatebar(T)
-		return
+	var/volume_left = volume
+	while(volume_left >= 5)
```

Comment:
I don't like this at all--and it certainly isn't consistent with any of the other reaction turfs. Yes, it means you can multiple the chocolate reagent with smoke, but that's really not an issue as chocolate is just sugar 2.0.